File: crates/wbw_r/examples/vector_topojson_roundtrip.R
# Simple TopoJSON roundtrip example for WbW-R.

library(whiteboxworkflows)

src <- "input_roads.gpkg"
topo <- "output_roads.topojson"
back <- "output_roads_back.gpkg"

roads <- wbw_read_vector(src)
wbw_write_vector(roads, topo)

roads_topo <- wbw_read_vector(topo)
wbw_write_vector(roads_topo, back)

cat(sprintf("wrote TopoJSON: %s\n", topo))
cat(sprintf("wrote roundtrip GeoPackage: %s\n", back))

File: crates/wbw_python/examples/vector_topojson_roundtrip.py
"""Simple TopoJSON roundtrip example for WbW-Py.

This script reads a vector layer, writes it as TopoJSON, and reads it back.
"""

from pathlib import Path
import whitebox_workflows as wb


def main() -> None:
    wbe = wb.WbEnvironment()

    src = Path("input_roads.gpkg")
    topo = Path("output_roads.topojson")
    back = Path("output_roads_back.gpkg")

    roads = wbe.read_vector(str(src))
    wbe.write_vector(roads, str(topo))

    roads_topo = wbe.read_vector(str(topo))
    wbe.write_vector(roads_topo, str(back))

    print(f"wrote TopoJSON: {topo}")
    print(f"wrote roundtrip GeoPackage: {back}")


if __name__ == "__main__":
    main()
